### PR TITLE
Deprecate several commands [CON-3591]

### PR DESCRIPTION
**Pull request by dustinmm80**
You can run `./deprecations.sh` to see all commands deprecated. This is its current output:

```
Planned deprecations for Conjur CLI
-----
group
    create    - Create a new group [DEPRECATED]
    retire    - Decommission a group [DEPRECATED]
    update    - Update group's attributes (eg. gidnumber) [DEPRECATED]
group members
    add    - Add a new group member [DEPRECATED]
    remove - Remove a group member [DEPRECATED]
hostfactory
    create - Create a new host factory [DEPRECATED]
host
    create         - Create a new host [DEPRECATED]
    enroll         - Enroll a new host into conjur [DEPRECATED]
    retire         - Decommission a host [DEPRECATED]
    update         - Update a hosts's attributes [DEPRECATED]
layer
    create    - Create a new layer [DEPRECATED]
    retire    - Decommission a layer [DEPRECATED]
layer hosts
    add             - Add a host to an layer [DEPRECATED]
    deny            - Remove a privilege on hosts in the layer [DEPRECATED]
    permit          - Permit a privilege on hosts in the layer [DEPRECATED]
    remove          - Remove a host from an layer [DEPRECATED]
resource
    annotate        - Set an annotation on a resource [DEPRECATED]
    create          - Create a new resource [DEPRECATED]
    deny            - Deny a privilege on a resource [DEPRECATED]
    give            - Grant ownership on a resource to a new owner [DEPRECATED]
    permit          - Give a privilege on a resource [DEPRECATED]
role
    create      - Create a new role [DEPRECATED]
    grant_to    - Grant a role to another role. You must have admin permission on the granting role. [DEPRECATED]
    revoke_from - Revoke a role from another role. You must have admin permission on the revoking role. [DEPRECATED]
user
    create          - Create a new user [DEPRECATED]
    retire          - Decommission a user [DEPRECATED]
    update          - Update a user's attributes [DEPRECATED]
variable
    create      - Create and store a variable [DEPRECATED]
    retire      - Decommission a variable [DEPRECATED]
```

### CODE REVIEW

File: lib/conjur/command/hosts.rb [hunk 1]
```diff
@@ -153,7 +159,7 @@ def self.host_layer_roles host
       end
     end
 
-    hosts.desc "[Deprecated] Enroll a new host into conjur"
+    hosts.desc "Enroll a new host into conjur [DEPRECATED]"
```
Comment (apotterri) [inline lib/conjur/command/hosts.rb]: Seems like the `action` block for this command should either call `notify_deprecated`, or have a comment explaining why it doesn't need to. Right now it looks like a mistake.

**Comment by apotterri** [comment]
Will you include the id of the story in the title (or the description), so it links to the JIRA issue, please?

**Comment by dustinmm80** [comment]
Yep, done.

**Comment by apotterri** [comment]
Needs a version bump and a CHANGELOG entry, too.

File: lib/conjur/command/hosts.rb (hunk 1 above)
Comment (dustinmm80) [inline lib/conjur/command/hosts.rb]: I skipped adding it here because it's already using `hide_docs(c)`, just normalized the DEPRECATED text

Comment (apotterri) [inline lib/conjur/command/hosts.rb]: Ok. I think it's worth adding a comment to that effect. Otherwise, it looks like it was forgotten.

**Pull request merged**